src/parquet/column_reader.h

```diff
@@ -123,7 +124,7 @@ class TypedColumnReader : public ColumnReader {
   typedef typename type_traits<TYPE>::value_type T;
 
   TypedColumnReader(const parquet::ColumnMetaData* metadata,
-      const parquet::SchemaElement* schema, InputStream* stream) :
+      const parquet::SchemaElement* schema, std::unique_ptr<InputStream>& stream) :
```

Comment:
I think it would be better to pass by value here, otherwise you can only pass an l-value as the `stream` argument:

http://stackoverflow.com/questions/8114276/how-do-i-pass-a-unique-ptr-argument-to-a-constructor-or-a-function